engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/QueryStringPredicateOptionsStep.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.predicate.dsl;
+
+import java.util.function.Consumer;
+
+import org.hibernate.search.engine.search.common.Rewrite;
+
+/**
+ * The final step in a "query string" predicate definition, where optional parameters can be set.
+ *
+ * @param <S> The "self" type (the actual exposed type of this step).
+ */
+public interface QueryStringPredicateOptionsStep<S extends QueryStringPredicateOptionsStep<?>>
+		extends CommonQueryStringPredicateOptionsStep<S> {
+
+	/**
+	 * Specifies whether * and ? are allowed as first characters of a search term. Default is true.
+	 * @param allowLeadingWildcard
+	 * @return {@code this}, for method chaining.
+	 */
+	S allowLeadingWildcard(boolean allowLeadingWildcard);
+
+	//	/**
+	//	 * Specifies whether backend should attempt to analyze wildcard terms.
+	//	 * Default is false.
+	//	 *
+	//	 * @param analyzeWildcard
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S analyzeWildcard(boolean analyzeWildcard);
```

Comment:
Seems useful, but:

1. Please be careful to use a default that matches that of the `wildcard` predicate.
2. Please give up if it's not already implemented for Lucene.

Reply:
ok, yeah looks like it needs some work for Lucene... so:

>I'd wait for an actual feature request before working on this. 

😃